sys/include/net/gnrc/rpl.h

```diff
@@ -627,6 +628,13 @@ extern netstats_rpl_t gnrc_rpl_netstats;
 #define CONFIG_GNRC_RPL_PARENT_TIMEOUT_DIS_RETRIES (3)
 #endif
 
+/**
+ * @brief Timeout for floating DODAGs in milliseconds.
+ */
+#ifndef CONFIG_GNRC_RPL_DODAG_FLOAT_TIMEOUT
+#  define CONFIG_GNRC_RPL_DODAG_FLOAT_TIMEOUT (15 * MS_PER_SEC * SEC_PER_MIN)
```

Comment:
15 minutes seems like an awful lot of time - where does that come from?

Reply:
During these 15 minutes, the node will maintain a floating DODAG for its subtree.   
That way nodes in the subtree can still reach each other, but there is no uplink. If at any point in time the node (or any child) receives a DIO from a grounded (non-floating) DODAG, it will leave the floating DODAG and switch to the grounded one.

If the 15min timeout is reached the floating DODAG will be deconstructed and cleared from the node and all its children.